### ISSUE TITLE
Bad dunder name is flagging __index__

### ISSUE THREAD

**Issue by Redoubts** [pylint-dev/pylint#8613]
### Bug description

consider 
```
class X:
    def __index__(self):
        return 1
```

this will flag [__index__](https://docs.python.org/3/library/operator.html#operator.__index__) as a bad dunder, even though it is valid to do.

### Command used

```shell
pylint x.py
```

### Pylint output

```shell
************* Module x
x.py:2:4: W3201: Bad or misspelled dunder method name __index__. (bad-dunder-name)
```

### Expected behavior

no lint

### Pylint version

```shell
% pylint --version
pylint 2.17.2
astroid 2.15.4
Python 3.10.11 (main, Apr 10 2023, 08:08:57) [GCC 8.3.0]
```

**Comment by Redoubts** [comment]
there's an obvious workaround in the config, but I think that shouldn't be needed

**Comment by Redoubts** [comment]
But I’m seeing this on 2.17?
